pygmt/tests/test_grdcut.py

```diff
@@ -43,21 +43,30 @@ def test_grdcut_file_in_dataarray_out():
 
 def test_grdcut_dataarray_in_file_out():
     "grdcut an input DataArray, and output to a grid file"
-    # Not supported yet.
-    # See https://github.com/GenericMappingTools/gmt/pull/3532
+    ingrid = load_earth_relief()
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        result = grdcut(ingrid, outgrid=tmpfile.name, region="0/180/0/90")
+        assert result is None  # grdcut returns None if output to a file
+        result = grdinfo(tmpfile.name, C=True)
+        assert result == "0 180 0 90 -8182 5651.5 1 1 180 90 1 1\n"
 
 
 def test_grdcut_dataarray_in_dataarray_out():
-    "grdcut an input DataArray, and output to a grid file"
-    # Not supported yet.
-    # See https://github.com/GenericMappingTools/gmt/pull/3532
-
-
-def test_grdcut_dataarray_in_fail():
-    "Make sure that grdcut fails correctly if DataArray is the input grid"
-    with pytest.raises(NotImplementedError):
-        grid = load_earth_relief()
-        grdcut(grid, region="0/180/0/90")
+    "grdcut an input DataArray, and output as DataArray"
+    ingrid = load_earth_relief()
```

Comment:
I would just use `grid = load_earth_relief()`, and maybe use a fixture so we don't load the same thing twice to speed up the test. You can follow this example:

https://github.com/GenericMappingTools/pygmt/blob/9a4c0a0177dac03e05d219c01105f272c9911655/pygmt/tests/test_grdimage.py#L13-L16

Reply:
Great. Will follow your example and merge it.

Good luck with your thesis!